component/client/chat/ClientChat.tsx

```diff
@@ -48,15 +49,15 @@ class ClientChat extends React.Component<IClientChatProps, IClientChatState> {
       messages : []
     };
     room.setUrl(url, (data: IData, user) => {
-      this.setState((prevState, props) => {
+      this.setState((prevState: IClientChatState, props: IClientChatProps) => {
```

Comment:
`userID` is for context menu control. I compare them so that I can either enable to disable the context menu. This was discussed with @zhoutwo in the group chat as the best alternative. 

The use case it came up in is changing of `username`. Previously, we could disallow the context menu from popping up by checking if the author of the message was the same as the current user; however, what if the current user changes their `username`. In this case, if they had previously written a message under the name `firstName` and changed their `username` to `secondName`, they would be able to open the context menu on the message they had written when their username was `firstName` which was something we wanted to avoid.